### ISSUE TITLE
systemd-resolved stops working when systemd-networkd is restartet

### ISSUE THREAD

**Issue by debug-richard** [systemd/systemd#25946]
### systemd version the issue has been seen with

247.3-7+deb11u1

### Used distribution

Debian 11

### Linux kernel version used

5.15.32

### CPU architectures issue was seen on

aarch64

### Component

systemd-resolved

### Expected behaviour you didn't see

DNS resolution works.

I didn't find any commits for this issue, so I'm posting it despite the old version.

### Unexpected behaviour you saw

DNS resolution stops working after restarting systemd-networkd.
Programs return "Temporary failure in name resolution" errors.

There is no log error/activity/output when this happens for systemd-networkd.

### Steps to reproduce the problem

Working:
```
$ resolvectl 
Global
       Protocols: +LLMNR +mDNS -DNSOverTLS DNSSEC=no/unsupported
resolv.conf mode: stub

Link 2 (enx)
Current Scopes: DNS LLMNR/IPv4 LLMNR/IPv6
     Protocols: +DefaultRoute +LLMNR -mDNS -DNSOverTLS DNSSEC=no/unsupported
   DNS Servers: 192.168.1.1
    DNS Domain: lan

Link 4 (docker0)
Current Scopes: none
     Protocols: -DefaultRoute +LLMNR -mDNS -DNSOverTLS DNSSEC=no/unsupported

```
Failure (Temporary failure in name resolution):
```
$ systemctl restart systemd-networkd
$ resolvectl
Global
       Protocols: +LLMNR +mDNS -DNSOverTLS DNSSEC=no/unsupported
resolv.conf mode: stub

Link 2 (enx)
Current Scopes: LLMNR/IPv4 LLMNR/IPv6
     Protocols: -DefaultRoute +LLMNR -mDNS -DNSOverTLS DNSSEC=no/unsupported

Link 4 (docker0)
Current Scopes: none
     Protocols: -DefaultRoute +LLMNR -mDNS -DNSOverTLS DNSSEC=no/unsupported
```
Working:
```
$ systemctl restart systemd-resolved
$ resolvectl
Global
       Protocols: +LLMNR +mDNS -DNSOverTLS DNSSEC=no/unsupported
resolv.conf mode: stub

Link 2 (enx)
Current Scopes: DNS LLMNR/IPv4 LLMNR/IPv6
     Protocols: +DefaultRoute +LLMNR -mDNS -DNSOverTLS DNSSEC=no/unsupported
   DNS Servers: 192.168.1.1
    DNS Domain: lan

Link 4 (docker0)
Current Scopes: none
     Protocols: -DefaultRoute +LLMNR -mDNS -DNSOverTLS DNSSEC=no/unsupported

```

**Comment by poettering** [comment]
*[Canned reply follows]*

This is the upstream bug and feature request tracker of systemd. Please use this only for issues in the two most current upstream systemd versions. See this link for the list of current releases.

https://github.com/systemd/systemd/releases

For support for older versions please contact your distribution instead.

In order to keep frustration at a minimum the submission form tried to make this all very clear.

**Comment by mkresin** [comment]
FTR

Hit the same bug with ubuntu 22.04 (systemd 249.11).

The issue is fixed by #21361, which is included in system 250+.
